### DISCUSSION TITLE
"sudo brew services start" has file ownership issue

### DISCUSSION THREAD

**Discussion by rucotec** [Homebrew/discussions#234] [Everyday usage]
I run homebrew under my user id (with admin privs). Thus all homebrew files belong to me. sudo brew services start creates a plist in /Library/LaunchDeamon, fine. But macOS on (re)boot of the system runs all these plist as root. The problem now is that such processes fail because the process executables are not owned by root but by my account. There seem to be two ways to solve this: change ownership of (at least certain) files installed by homebrew or edit the plist and enter a userName key so that things do not run as root upon boot. Both have their drawbacks. Changing the file ownership prevents homebrew from updating as it cannot replace files, editing the plist is tedious and dangerous and maybe I do not want to run systemwide stuff under my uid.

I am looking for a strategy to handle this. Any ideas?

**Comment by SMillerDev** [comment]
@MikeMcQuaid didn't we chown the files when running services as `sudo`?

**Reply by MikeMcQuaid** [reply]
Yes, we do that because it's a security vulnerability to not do so. You'll need to `sudo brew services stop` (which fixes the permissions) before upgrading.
